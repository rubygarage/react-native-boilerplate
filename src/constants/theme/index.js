export default {
  colors: {
    black: '#000000',
    white: '#ffffff',
  },
  fonts: {
  },
  sizes: {
  },
};
