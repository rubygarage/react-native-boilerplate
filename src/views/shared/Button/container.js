import { withTheme } from 'styled-components/native';

import ButtonComponent from './component';

export default withTheme(ButtonComponent);
