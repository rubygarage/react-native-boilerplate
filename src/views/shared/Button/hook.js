import { useContext } from 'react';
import { ThemeContext } from 'styled-components';

const useContainer = () => {
  const theme = useContext(ThemeContext);

  return { theme };
};

export default useContainer;
