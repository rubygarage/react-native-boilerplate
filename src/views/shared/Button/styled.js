import styled from 'styled-components/native';

const Wrap = styled.TouchableOpacity`
  min-width: ${(props) => (props.isSmallWidth ? 72 : 136)};
  height: ${(props) => (props.isSmallHeight ? '36px' : '48px')};
  justify-content: center;
  align-items: center;
  padding-right: 8;
  padding-left: 8;
  border-radius: 4;
  flex-shrink: 0;
  background-color: ${({ theme, isOutline, fillColor }) => (isOutline ? theme.colors.white : fillColor)};
  margin-right: ${(props) => (props.isMarginRight ? '16' : '0')};
  margin-bottom: ${(props) => (props.isMarginBottom ? '16' : '0')};
  border-width: ${(props) => (props.isBorder ? '2' : '0')};
  border-style: solid;
  border-color: ${({ theme }) => theme.colors.black};
`;

const Text = styled.Text`
  color: ${({ theme, isOutline }) => (isOutline ? theme.colors.black : theme.colors.white)};
  font-weight: bold;
`;

export { Wrap, Text };
