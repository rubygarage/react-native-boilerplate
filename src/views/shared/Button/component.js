import React from 'react';
import PropTypes from 'prop-types';
import { ActivityIndicator, ViewPropTypes } from 'react-native';

import * as S from './styled';

const Button = ({
  children, isOutline, isLoading, isBorder, disabled,
  isMarginRight, isMarginBottom, isSmallWidth,
  onPress, theme, isSmallHeight, fillColor = theme.colors.blue200,
  style,
}) => (
  <S.Wrap
    style={style}
    isOutline={isOutline}
    isBorder={isBorder}
    isMarginRight={isMarginRight}
    isMarginBottom={isMarginBottom}
    isSmallWidth={isSmallWidth}
    isSmallHeight={isSmallHeight}
    onPress={onPress}
    fillColor={fillColor}
    disabled={disabled}
  >
    {isLoading
      ? <ActivityIndicator size="large" color={theme.colors.white} />
      : <S.Text isOutline={isOutline} fillColor={fillColor}>{children}</S.Text>}
  </S.Wrap>
);

Button.propTypes = {
  children: PropTypes.node.isRequired,
  theme: PropTypes.shape({ colors: PropTypes.shape().isRequired }).isRequired,
  onPress: PropTypes.func,
  isLoading: PropTypes.bool,
  isOutline: PropTypes.bool,
  isBorder: PropTypes.bool,
  isMarginRight: PropTypes.bool,
  isMarginBottom: PropTypes.bool,
  isSmallWidth: PropTypes.bool,
  isSmallHeight: PropTypes.bool,
  disabled: PropTypes.bool,
  fillColor: PropTypes.string,
  style: ViewPropTypes.style,
};

Button.defaultProps = {
  onPress: undefined,
  isLoading: false,
  isOutline: false,
  isBorder: false,
  isMarginRight: false,
  isMarginBottom: false,
  isSmallWidth: false,
  isSmallHeight: false,
  disabled: false,
  fillColor: undefined,
  style: undefined,
};

export default Button;
