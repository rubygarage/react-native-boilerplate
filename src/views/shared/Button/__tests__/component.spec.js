import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';

import theme from 'utils/testHelpers/mockedTheme';
import Button from '../component';

const onPress = jest.fn();
const defaultProps = {
  onPress, isLoading: false, isSmall: undefined, isOutline: undefined, theme,
};

describe('Button component', () => {
  const component = shallow(<Button {...defaultProps}><Text>Test Child</Text></Button>);

  it('renders correctly default', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with enabled styles', () => {
    const props = {
      ...defaultProps,
      isSmall: true,
      isOutline: true,
      isMargin: true,
    };
    component.setProps(props);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with loading state', () => {
    const props = {
      ...defaultProps,
      isLoading: true,
    };
    component.setProps(props);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with enabled styles and fillColor', () => {
    const props = {
      ...defaultProps,
      isSmall: true,
      isOutline: true,
      isMargin: true,
      fillColor: theme.colors.green500,
    };
    component.setProps(props);
    expect(component).toMatchSnapshot();
  });
});
