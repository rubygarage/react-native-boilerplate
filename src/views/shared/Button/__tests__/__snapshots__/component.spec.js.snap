// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly default 1`] = `
<Styled(TouchableOpacity)
  disabled={false}
  fillColor="[mocked blue200]"
  isBorder={false}
  isMarginBottom={false}
  isMarginRight={false}
  isOutline={false}
  isSmallHeight={false}
  isSmallWidth={false}
  onPress={[MockFunction]}
>
  <Styled(Text)
    fillColor="[mocked blue200]"
    isOutline={false}
  >
    <Text>
      Test Child
    </Text>
  </Styled(Text)>
</Styled(TouchableOpacity)>
`;

exports[`Button component renders correctly with enabled styles 1`] = `
<Styled(TouchableOpacity)
  disabled={false}
  fillColor="[mocked blue200]"
  isBorder={false}
  isMarginBottom={false}
  isMarginRight={false}
  isOutline={true}
  isSmallHeight={false}
  isSmallWidth={false}
  onPress={[MockFunction]}
>
  <Styled(Text)
    fillColor="[mocked blue200]"
    isOutline={true}
  >
    <Text>
      Test Child
    </Text>
  </Styled(Text)>
</Styled(TouchableOpacity)>
`;

exports[`Button component renders correctly with enabled styles and fillColor 1`] = `
<Styled(TouchableOpacity)
  disabled={false}
  fillColor="[mocked green500]"
  isBorder={false}
  isMarginBottom={false}
  isMarginRight={false}
  isOutline={true}
  isSmallHeight={false}
  isSmallWidth={false}
  onPress={[MockFunction]}
>
  <Styled(Text)
    fillColor="[mocked green500]"
    isOutline={true}
  >
    <Text>
      Test Child
    </Text>
  </Styled(Text)>
</Styled(TouchableOpacity)>
`;

exports[`Button component renders correctly with loading state 1`] = `
<Styled(TouchableOpacity)
  disabled={false}
  fillColor="[mocked blue200]"
  isBorder={false}
  isMarginBottom={false}
  isMarginRight={false}
  isOutline={false}
  isSmallHeight={false}
  isSmallWidth={false}
  onPress={[MockFunction]}
>
  <ActivityIndicator
    animating={true}
    color="[mocked white]"
    hidesWhenStopped={true}
    size="large"
  />
</Styled(TouchableOpacity)>
`;
